addons/sales_team/models/crm_team_member.py

```diff
@@ -50,10 +54,89 @@ def _compute_crm_team_skip_ids(self):
         for member in self:
-            if member.crm_team_id:
+            if member_user_ids:
+                member.user_in_teams_ids = member_user_ids
+            elif member.crm_team_id:
                 member.user_in_teams_ids = member.crm_team_id.member_ids
             elif self.env.context.get('default_crm_team_id'):
                 member.user_in_teams_ids = self.env['crm.team'].browse(self.env.context['default_crm_team_id']).member_ids
             else:
                 member.user_in_teams_ids = self.env['res.users']
+
+    @api.depends('crm_team_id')
+    def _compute_is_membership_multi(self):
+        multi_enabled = self.env['ir.config_parameter'].sudo().get_param('sales_team.membership_multi', False)
+        for member in self:
+            member.is_membership_multi = multi_enabled
+
+    @api.depends('user_id', 'crm_team_id')
+    def _compute_is_in_another_team(self):
+        existing = self.env['crm.team.member'].search([('user_id', 'in', self.user_id.ids)])
+        user_mapping = dict.fromkeys(existing.user_id, self.env['crm.team'])
+        for membership in existing:
+            user_mapping[membership.user_id] |= membership.crm_team_id
+        for member in self:
+            if not user_mapping.get(member.user_id):
+                member.is_in_another_team = False
+                continue
+            teams = user_mapping[member.user_id]
+            remaining = teams - (member.crm_team_id | member._origin.crm_team_id)
+            member.is_in_another_team = len(remaining) > 0
+
+    @api.model_create_multi
+    def create(self, values_list):
+        """ Specific behavior implemented on create
+
+          * mono membership mode: other user memberships are automatically
+            archived (a warning already told it in form view);
+          * creating a membership already existing as archived: old one is
+            automatically removed;
+        """
+        is_membership_multi = self.env['ir.config_parameter'].sudo().get_param('sales_team.membership_multi', False)
+        exist_all = self.with_context(active_test=False).search([
+            ('user_id', 'in', [values['user_id'] for values in values_list])
+        ])
+        user_memberships = dict.fromkeys(exist_all.user_id.ids, self.env['crm.team.member'])
+        for membership in exist_all:
+            user_memberships[membership.user_id.id] += membership
+
+        exist_to_remove = self.env['crm.team.member']
+        exist_to_archive = self.env['crm.team.member']
+        for values in values_list:
+            exist_current = user_memberships.get(values['user_id'])
+            if not exist_current:
+                continue
+            if not is_membership_multi:
+                to_archive = exist_current.filtered(lambda m: m.active and m.crm_team_id.id != values['crm_team_id'])
+                if to_archive:
+                    exist_to_archive += to_archive
```

Comment:
Can't you do:
`exist_to_archive += exist_current.filtered(lambda m: m.active and m.crm_team_id.id != values['crm_team_id'])`

Directly?
(Same for remove here under)

Reply:
Code simplified as a whole.